src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.WorkerThread.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using Internal.Runtime.Augments;
+
+namespace System.Threading
+{
+    internal static partial class ClrThreadPool
+    {
+        internal static class WorkerThread
+        {
+            internal static WaitSubsystem.WaitableObject s_semaphore = WaitSubsystem.WaitableObject.NewSemaphore(0, int.MaxValue);
+
+            private const int TimeoutMs = 20 * 1000;
+            
+            private static void WorkerThreadStart()
+            {
+                RuntimeThread.InitializeThreadPoolThread();
+                // TODO: Event: Worker Thread Start event
+
+                while (true)
+                {
+                    // TODO: Event:  Worker thread wait event
+                    while (WaitSubsystem.Wait(s_semaphore, TimeoutMs, false, true))
+                    {
+                        if(!ThreadPoolWorkQueue.Dispatch())
+                        {
+                            return;
+                        }
+
+                        RuntimeThread.CurrentThread.Priority = ThreadPriority.Normal;
```

Comment:
- RuntimeThread.CurrentThread involves a thread-local lookup and should be cached at the top of the function
- Since a lock is involved in setting the priority, it would be better to check it before setting it